packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

```diff
@@ -282,6 +282,43 @@ void main() {
       Platform: () => fakePlatform,
     });
 
+    testUsingContext('Show current version to the upgrade message.', () async {
+      const String revision = 'abc123';
+      const String upstreamRevision = 'def456';
+      const String version = '1.2.3';
+      const String upstreamVersion = '4.5.6';
+
+      when(flutterVersion.frameworkRevision).thenReturn(revision);
+      when(flutterVersion.frameworkRevisionShort).thenReturn(revision);
+      when(flutterVersion.frameworkVersion).thenReturn(version);
+
+      final MockFlutterVersion latestVersion = MockFlutterVersion();
+
+      when(latestVersion.frameworkRevision).thenReturn(upstreamRevision);
+      when(latestVersion.frameworkRevisionShort).thenReturn(upstreamRevision);
+      when(latestVersion.frameworkVersion).thenReturn(upstreamVersion);
+
+      fakeCommandRunner.alreadyUpToDate = false;
+      fakeCommandRunner.remoteVersion = latestVersion;
+
+      processManager.addCommand(
+        FakeCommand(
+            command: <String>[
+              globals.fs.path.join('bin', 'flutter'),
+              'upgrade',
+              '--continue',
+              '--no-version-check',
+            ],
+            environment: <String, String>{'FLUTTER_ALREADY_LOCKED': 'true', ...fakePlatform.environment}
+        ),
+      );
+      realCommandRunner.recordState(flutterVersion, latestVersion);
+      expect(testLogger.statusText, contains('Upgrading Flutter to ${latestVersion.frameworkVersion} from ${flutterVersion.frameworkVersion} in ${realCommandRunner.workingDirectory}'));
```

Comment:
Tests should be matched against hardcoded strings.
See https://github.com/flutter/flutter/pull/69420#discussion_r519069628 for context.